create_builder.go

```diff
@@ -102,10 +105,22 @@ func (c *Client) CreateBuilder(ctx context.Context, opts CreateBuilderOptions) e
 				return errors.Wrapf(err, "downloading buildpack from %s", style.Symbol(b.URI))
 			}
 
+			isOCILayout, err := buildpackage.IsOCILayoutBlob(blob)
+			if err != nil {
+				return errors.Wrap(err, "inspecting buildpack blob")
+			}
+
+			if isOCILayout {
+				// The buildpack is a Package, and should be added in the next section
```

Comment:
I wonder if this is a bit fragile - if for some reason we decided to iterate through `Packages()` first instead of `Buildpacks()`, we would miss .cnb files. Maybe it's worth adding a comment before line 138 explaining why that section needs to come second?

Reply:
I agree, which is why I resolved the two loops into one. Thanks for calling it out!